code-studio/scripts/build-js.js

```diff
@@ -100,7 +102,30 @@ Promise.all([
   }
 
   if (commander.watch) {
-    console.log("Watching for changes...");
+    // Watching for JavaScript file changes is already set up by browserify/watchify on each bundle
+    console.log("Watching for .js file changes...");
+
+    // Use gaze to set a watcher for scss file changes
+    gaze('src/css/*.scss', function (err, watcher) {
+
+      console.log("Watching for .scss file changes...");
+
+      /* Uncomment this if you want to debug .scss watching
+
+      console.log("cwd: " + process.cwd());
+      var watched = this.watched();
+      Object.keys(watched).forEach(function(watchedDir) {
+        watched[watchedDir].forEach(function(watchedFile) {
+          console.log('Watching ' + watchedFile + ' for changes.');
+        });
+      });
+      */
+
+      this.on('changed', function (filepath) {
```

Comment:
It's not immediately obvious what `this` represents here.  Looking at the docs for gaze, I think you could use the `watcher` argument passed to this callback instead.  That might be easier to read.